websites/X/Xoul - AI/presence.ts

```diff
@@ -0,0 +1,112 @@
+const presence = new Presence({
+		clientId: "1310165299802411078",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/aTWK1eZ.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		[time, buttons] = await Promise.all([
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		button = {
+			label: "View Page",
+			url: href,
+		};
```

Comment:
this is not allowed, since u can't have button to the homepage.
```suggestion
		button = {
			label: "View Page",
			url: href,
		};
```

Reply:
I removed the button from homepage.

Reply:
`View Page` is somewhat generic. It would be better to place more specific button messages for the various paths. The buttons should also lead to places that a user might want to visit (profile page, project page, etc.).

Reply:
Makes sense, will modify it in a commit today, is there anything else that you would like to have changed?